File: backend/Models/Cabin.cs
namespace CabinConnect.Api.Models;

public class Cabin
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public string[] Amenities { get; set; } = [];
    public Guid OwnerId { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: src/web/src/types/cabin.ts
/** Shared response types matching the .NET API DTO shapes agreed in elaboration. */

export interface Community {
  id: string
  name: string
  region: string
}

export interface Amenity {
  id: string
  name: string
}

export interface CabinAddress {
  street: string
  postalCode: string
  city: string
  country: string
}

export interface Cabin {
  id: string
  name: string
  address: CabinAddress
  community: Community
  capacity: number
  amenities: Amenity[]
  createdAt: string
  updatedAt: string
}

export interface CreateCabinRequest {
  name: string
  address: CabinAddress
  communityId: string
  capacity: number
  amenityIds: string[]
}

export interface UpdateCabinRequest {
  name?: string
  address?: Partial<CabinAddress>
  communityId?: string
  capacity?: number
  amenityIds?: string[]
}

export interface ValidationProblem {
  title: string
  errors: Record<string, string[]>
}
